qiime2/sdk/plugin_manager.py

```diff
@@ -100,11 +101,28 @@ def _integrate_plugin(self, plugin):
 
             self.formats[name] = record
         self.type_formats.extend(plugin.type_formats)
+        self.importable_types.update(plugin.importable_types)
 
     # TODO: Should plugin loading be transactional? i.e. if there's
     # something wrong, the entire plugin fails to load any piece, like a
     # databases rollback/commit
 
+    @property
+    def get_semantic_types(self, *, plugin=None):
+        from qiime2.plugin import SemanticTypeRecord
+
+        types = set()
+
+        plugins = [plugin] if plugin else self.plugins
+
+        for plugin in plugins:
+            for name, record in plugin.type_fragments.items():
```

Comment:
Please refactor this to grab the `SemanticTypeRecord`s directly off of the plugin manager's `self.types` property. Since we discussed pushing this logic into `plugin.py` above, the records will already be created and ready for use by the time you get here. Please let me know if you need any clarification!

:electric_plug: